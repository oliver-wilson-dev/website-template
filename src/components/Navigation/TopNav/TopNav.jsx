import React from 'react';
import { NavLink } from 'react-router-dom';
import routes from '../../../routes';

import styles from './TopNav.module.css';

const { home, about, contact } = routes;

const TopNav = () => (
  <nav className={styles.navigation}>
    <ul className={styles.navItems}>
      <li className={styles.navItem}>
        <NavLink className={styles.link} exact activeClassName={styles['link--active']} to={home.route}>Home</NavLink>
      </li>
      <li className={styles.navItem}>
        <NavLink className={styles.link} activeClassName={styles['link--active']} to={about.route}>About</NavLink>
      </li>
      <li className={styles.navItem}>
        <NavLink className={styles.link} activeClassName={styles['link--active']} to={contact.route}>Contact</NavLink>
      </li>
    </ul>
  </nav>
);

export default TopNav;
