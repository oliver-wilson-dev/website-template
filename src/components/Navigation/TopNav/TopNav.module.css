@import "../../../styles/globalVars.css";

.navigation {
  display: none;
  height: 100%;
}

.navItems {
  display: flex;
  padding: 0px;
  margin: 0px;
  list-style-type: none;
  justify-content: center;
  align-items: center;
  flex-grow: 1;
  cursor: pointer;
  max-width: $max-content-width;
  height: 100%;
  -webkit-tap-highlight-color: transparent;
}

.navItem {
  display: flex;
  height: 100%;
  margin-right: 5px;

  &:last-child {
    margin-right: 0px;
  }
}

.link {
  padding: 15px 30px;
  padding: 15px;
  text-decoration: none;
  color: #000000;
  transition: background-color 0.4s, opacity 0.4s;

  &:hover {
    background-color: #fcbf49;
  }

  &--active {
    background-color: #fcbf49;

    &:hover {
      opacity: 0.5;
    }
  }
}

@media (min-width: 768px) {
  .link {
    padding: 15px 30px;
  }

  .navItems {
    justify-content: flex-end;
  }

  .navigation {
    display: block;
  }
}
