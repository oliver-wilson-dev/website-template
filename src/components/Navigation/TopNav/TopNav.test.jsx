import React from 'react';
import { shallow } from 'enzyme';
import TopNav from './TopNav';

jest.mock('../../../routes', () => ({
  home: {
    route: 'test-home-route'
  },
  about: {
    route: 'test-about-route'
  },
  contact: {
    route: 'test-contact-route'
  }
}));

jest.mock('../../../containers/ThemeSwitch', () => {
  const ThemeSwitch = (props) => <button type="button" {...props} />;

  return ThemeSwitch;
});

jest.mock('react-router-dom', () => ({
  NavLink: ({ children }) => <>{children}</>
}));

const render = () => shallow(<TopNav />);

it('renders correctly', () => {
  expect(render()).toMatchSnapshot();
});
