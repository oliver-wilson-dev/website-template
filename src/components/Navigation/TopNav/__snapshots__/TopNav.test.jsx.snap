// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navigation"
>
  <ThemeSwitch />
  <ul
    className="navItems"
  >
    <li
      className="navItem"
    >
      <NavLink
        activeClassName="link--active"
        className="link"
        exact={true}
        to="test-home-route"
      >
        Home
      </NavLink>
    </li>
    <li
      className="navItem"
    >
      <NavLink
        activeClassName="link--active"
        className="link"
        to="test-about-route"
      >
        About
      </NavLink>
    </li>
    <li
      className="navItem"
    >
      <NavLink
        activeClassName="link--active"
        className="link"
        to="test-contact-route"
      >
        Contact
      </NavLink>
    </li>
  </ul>
</nav>
`;
